ratis-common/src/main/java/org/apache/ratis/protocol/RaftGroup.java

```diff
@@ -75,6 +77,13 @@ public Collection<RaftPeer> getPeers() {
     return peers.values();
   }
 
+  public Collection<RaftPeer> getPeers(RaftProtos.RaftPeerRole role) {
+    return peers.values()
+        .stream()
+        .filter(peer -> peer.getStartupRole() == role)
```

Comment:
I think we can't use `startupRole` as the difference between raftpeer, because the peer will change its identity

Reply:
fix.